ansible/roles/postgresql/tasks/main.yml

```diff
@@ -98,11 +99,31 @@
     name: "{{ item.name }}"
     state: present
     port: "{{ postgresql_port }}"
-    owner: "{{ localsettings.PG_DATABASE_USER }}"
+    owner: "{{ item.owner }}"
   with_items: postgresql_dbs
   when:
     pg_install == True
 
+- name: Grant PostgreSQL privileges
+  sudo_user: postgres
+  postgresql_privs: type=database database="{{ item.1.db_name }}" roles="{{ item.0.username}}" privs="{{ item.1.priv}}"
+  with_subelements:
+    - "{{ postgres_users.values() }}"
+    - privileges
+    - flags:
+      skip_missing: True
+
+- name: Set readonly permissions
```

Comment:
I suspect we need to do something like this: https://gist.github.com/jirutka/afa3ce62b1430abf7572#file-pg_grant_read_to_db-sh

one issue though is that setting the default privileges needs to be done by the `commcarehq` user so I guess using the `shell` or `command` module is necessary

here is an eg where we do something similar: https://github.com/dimagi/commcarehq-ansible/blob/757d383105114e0c900c193a578664174b69bd5b/ansible/roles/postgresql/tasks/main.yml#L173-L173
 
GitHub
dimagi/commcarehq-ansible
Contribute to commcarehq-ansible development by creating an account on GitHub.